### DISCUSSION TITLE
Compiling DSPF with Local GIT Repository

### DISCUSSION THREAD

**Discussion by antoine59910** [codefori/vscode-ibmi#1794] [Q&A]
Hello,

I am currently working on a project, with my source code managed in a local GIT repository. I am seeking advice or best practices on how to compile a Display File (DSPF) within this environment. (local actions)

I would like to know the most efficient way to compile a DSPF when working locally with GIT or must i move the source to the server and compil it there ?

 Any insights, shared experiences, or recommendations on tools and workflows for this scenario would be greatly appreciated.

Thank you in advance for your assistance.

**Comment by chrjorgensen** [comment]
@antoine59910 

A DSPF has to be build on the server like all other IBM i objects. You will need to deploy (push) the sources from the local workspace to the server before compile is started - your local workspace decides how the push is done and how the objects are build.

For more information about local development, read this: https://codefori.github.io/docs/#/pages/developing/local/getting-started

The command CRTDSPF does not support streamfiles, which is what you have in your git repository. However, if you get the CRTFRMSTMF command installed on your server, you can use this to compile the streamfile anyway, by telling CRTFRMSTMF to issue the CRTDSPF command on the uploaded source.

**Comment by andreas-prouza** [comment]
In projects I am also working with local sources and git.
Specially if you have a large number of sources and if you also want to use branches in git, it's not like a simple deploy (push) of a source.

Therefore I started a concept which can handle all these topics and is very fast: 
https://github.com/andreas-prouza/ibm-i-build
